### PR TITLE
fix Easy Config-related cleanups #5388

### PR DESCRIPTION
**Pull request by Simulant87**
fixing #5388

### CODE REVIEW

**Comment by dmlloyd** [comment]
I suspect this will fail to compile as I don't see any changes where the usages of the modified fields are updated.  I'll check it over once the build is green.  But merging should wait until after #5387.

**Comment by Simulant87** [comment]
I was confused because, I could not find any usages of the configuration fields. Do I have to setup sources from additonal repositories?

**Comment by dmlloyd** [comment]
All the usages should be within the same source repository.  You might need to check to ensure that your IDE has correctly imported the project.

**Comment by Simulant87** [comment]
I found the callers and fixed the compile errors (hopefully all of them), but I didn't had IDE support for compile checks.

**Review by gsmet — CHANGES_REQUESTED** [review]
Let's wait for David's config PR to be in before moving on this.

All the fix commits must be squashed: we don't want commits that don't compile.

**Comment by machi1990** [comment]
@Simulant87 FYI the config PR #5387 is in. Can you please rebase with master to pick related updates? Thanks.

**Comment by Simulant87** [comment]
@machi1990 rebase is done.

**Comment by dmlloyd** [comment]
Looks like the rebase didn't work right.

**Comment by Simulant87** [comment]
I don't see what went wrong. Could you help me?

**Comment by machi1990** [comment]
> I don't see what went wrong. Could you help me?

I see that you have over 179 commits and lots of these changes are already in the master.

**Comment by Simulant87** [comment]
But I don't see additional changes in the diff and the branch should be sqash merged anyway.
How to proceed?

**Comment by machi1990** [comment]
> But I don't see additional changes in the diff and the branch should be sqash merged anyway.
> How to proceed?

Hi @Simulant87 I would either open another PR with a clean history  or optionally force push in the current one by following running the following commands:

- `git fetch https://github.com/quarkusio/quarkus master` 
- `git reset --hard FETCH_HEAD`
- `curl https://patch-diff.githubusercontent.com/raw/quarkusio/quarkus/pull/5489.diff > 5489.diff`
- `git apply 5489.diff`
- `git clean -df`
- `git commit -am "fix config-related cleanups #5388"`
- `git push -f`

HTH

**Comment by Simulant87** [comment]
@machi1990 thanks for providing the commands. I applyed them and the history looks good. 1 single commit.

**Review by machi1990 — CHANGES_REQUESTED** [review]
The history looks much better. Thanks @Simulant87 

Can we also deal with  
> @ConfigItem which is a List type that is always converted to an array using toArray can be changed to just be an array type instead.

e.g for [FlywayRunTimeConfiguration#schemas](https://github.com/quarkusio/quarkus/blob/bc066bcfc8c95833845bbb70ad75d61cba9ddafd/extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayRuntimeConfig.java#L25) and [FlywayBuildConfig#locations](https://github.com/quarkusio/quarkus/blob/bc066bcfc8c95833845bbb70ad75d61cba9ddafd/extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayBuildConfig.java#L20) we could probably have their type changed to `Optional<String[]>`

Thanks

**Review by dmlloyd — DISMISSED** [review]

**Comment by dmlloyd** [comment]
Looks OK to me, once everyone else's issues have been addressed.

**Comment by Simulant87** [comment]
@machi1990 please review the latest changes

File: extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java
Comment (machi1990) [inline extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java]: Is this check required?

Comment (Simulant87) [reply extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java]: I am not 100% sure. it was there before see old line 132 below

**Comment by machi1990** [comment]
> @machi1990 please review the latest changes

Thanks @Simulant87. Can we squash into one commit? 

I also have a general comment about the usage of lambdas in runtime module that I just noticed in your changes. Can we avoid it? 

Thank you.

File: extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java
Comment (Simulant87) [reply extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java]: if it is not possible get `null` from a Config, it could be removed.

**Comment by Simulant87** [comment]
@machi1990 I commented on the null-check and also think that it is not necessary, if you agree. I assume the git command you posted earlier would still squash my commit again without any changes required to the commands.

So I would remove the null-check, commit, push, execute the commands and then we have a clear history.

**Comment by machi1990** [comment]
> @machi1990 I commented on the null-check and also think that it is not necessary, if you agree. I assume the git command you posted earlier would still squash my commit again without any changes required to the commands.
> 
> So I would remove the null-check, commit, push, execute the commands and then we have a clear history.

Thanks. Yes, the above command will work but it is an overkill for a little number of commits you have. You can use a `git rebase` command instead if you are at ease with it.

File: extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java
Comment (dmlloyd) [reply extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java]: It is indeed not possible to get `null` from a `Config`.

**Comment by Simulant87** [comment]
@machi1990 : The pipeline is green, the history is clean. I think all change requests are addressed. Please have a final look and I hope you will merge it.

File: extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java
Comment (gsmet) [inline extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java]: I think I would keep this null check here just to be safe.

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java
Comment (gsmet) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java]: We try to avoid lambdas in runtime code as it consumes more memory than strictly necessary.

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/cors/CORSFilter.java
Comment (gsmet) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/cors/CORSFilter.java]: Same here.

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/PathMatchingHttpSecurityPolicy.java
Comment (gsmet) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/PathMatchingHttpSecurityPolicy.java]: Same here.

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks for this work @Simulant87 . I think it's nearly ready to be merged. I would like to do it just after I branched 1.1.0.

There are a couple of small things to fix in the PR though.

Could you have a look?

Thanks!

File: extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java
Comment (Simulant87) [reply extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java]: I already discussed with @machi1990 that the null check is not necessary. If you want to keep it please clarify it with him as you both reviewers have oposite opinions.

Also the old code was not written "save" but assumes that `location` is never `null`. Line 135 `....getResources(location);` is not protected by the null check above and asserts the parameter to be not null in the first line of the call.

**Comment by Simulant87** [comment]
I am loosing interest on keeping my branch up to date while waiting for the last review to be able to merge.

I also don't know how to resolve the conflicts which I am facing when I try to rebase on master. Some help here would be appreciated.

**Comment by machi1990** [comment]
> I am loosing interest on keeping my branch up to date while waiting for the last review to be able to merge.

Hi @Simulant87 sorry for the delay as I was mostly away for a month - festive period. 
 
> I also don't know how to resolve the conflicts which I am facing when I try to rebase on master. Some help here would be appreciated.

I planned to take care of the rebase but I cannot push to your branch. Can you please apply this patch
[5388.diff](https://github.com/quarkusio/quarkus/files/4079850/5388.log) using the commands below:

First rebase your work with master
```
git fetch https://github.com/quarkusio/quarkus master
git reset --hard FETCH_HEAD
```

Now download the patch from https://github.com/quarkusio/quarkus/files/4079850/5388.log and save it as `5388.diff` to Quarkus project dir. 

Then apply the patch and push the changes.
```
git apply 5388.diff
git clean -df
git commit -am "fix config-related cleanups #5388"
git push -f
```

Let me know if you encounter any problem.

**Comment by Simulant87** [comment]
Thanks @machi1990 for creating the patch. I applied it following your description.

File: extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java
Comment (machi1990) [inline extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java]: Oh my bad, I did not see this star import. We usually do not allow them in the codebase. Sorry I should have spotted it. 

Can you please deal import individual classes and get rid of it (making sure that it is into one commit) ? Thank you

**Review by machi1990 — CHANGES_REQUESTED** [review]
Hey @Simulant87 thank you for taking care of the rebase patch. I just spotted a small thing that I should have spotted earlier. It should be easy. Once that is dealt with I'll approve and ask Guillaume for e re-review before merging it. We are really there this time!

**Comment by machi1990** [comment]
@Simulant87 thanks 👌 Can you squash the two commits?

**Comment by machi1990** [comment]
@Simulant87 can we get rid of the merge commit please? Otherwise LGTM.

@gsmet were your comments addressed? I think once the merge commit is removed we can safely merge this one. WDYT?

**Review by machi1990 — APPROVED** [review]

File: extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/NettyHttpClientConfig.java
Comment (gsmet) [inline extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/NettyHttpClientConfig.java]: Same here.

File: extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java
Comment (gsmet) [reply extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java]: It doesn't look solved to me?

File: extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceBuildTimeConfig.java
Comment (gsmet) [inline extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceBuildTimeConfig.java]: Why is it Optional? Since we have a default value, it shouldn't be.

File: extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientBuildTimeConfig.java
Comment (gsmet) [inline extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientBuildTimeConfig.java]: Same here.

**Review by gsmet — CHANGES_REQUESTED** [review]
This generally looks good, thanks.

But I added a few comments. For the int default value, it's a matter of taste, I think being explicit has value here.

File: extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceBuildTimeConfig.java
Comment (dmlloyd) [reply extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceBuildTimeConfig.java]: Any property that can be empty has to be optional, regardless of whether the default is non-empty.  If the property cannot be empty, then it should not be optional.

File: extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceRuntimeConfig.java
Comment (dmlloyd) [reply extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceRuntimeConfig.java]: I suppose the place to discuss this would be #5388?  They have simply implemented the requirement.

**Comment by Simulant87** [comment]
You are nice guys, really helping me out and supporting. But the process to contribute is frustrating. Keeping the branch up to date, resolve merge conflicts together with the one commit convention, sometimes long review times and the most: open discussions about the implementation after a "good to go".

Therefore I will not update the branch any more. Take my code and do what ever you want with it, to get the changes you want into master. I don't care about attribution to simplify it. But I will not update the branch, wait for the next review and eventually have to do it 2-3 times again because of the resons stated above. Sorry for that.

**Comment by dmlloyd** [comment]
I can only agree.  This issue was meant to be simple to resolve, but it seems the simpler the PR, the more nitpicky we get with it.  This probably needs to be addressed somehow.

**Comment by machi1990** [comment]
Superseded by https://github.com/quarkusio/quarkus/pull/8497. 

Let's close this one.

Thanks @Simulant87 for giving this a shot and sorry for the delay it took to merge this one and the frustrations the review process might have caused you.

**Pull request closed without merging**
